backend/src/modules/teamusers/applications/add-and-remove-team-users.use-case.ts

```diff
@@ -0,0 +1,36 @@
+import TeamUserDto from 'src/modules/teamusers/dto/team.user.dto';
+import { TeamUserRepositoryInterface } from '../interfaces/repositories/team-user.repository.interface';
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+import { TYPES } from '../interfaces/types';
+import { AddAndRemoveTeamUsersUseCaseInterface } from '../interfaces/applications/add-and-remove-team-users.use-case.interface';
+import { UPDATE_FAILED } from 'src/libs/exceptions/messages';
+import TeamUser from 'src/modules/teams/entities/team.user.schema';
+import { CreateTeamUserServiceInterface } from '../interfaces/services/create.team.user.service.interface';
+import { DeleteTeamUserServiceInterface } from '../interfaces/services/delete.team.user.service.interface';
+
+@Injectable()
+export class AddAndRemoveTeamUsersUseCase implements AddAndRemoveTeamUsersUseCaseInterface {
```

Comment:
```suggestion
type Args = {
addUsers: TeamUserDto[]; removeUsers: string[];
}

export class AddAndRemoveTeamUsersUseCase implements UseCase<Args, TeamUser[]> {
```

You can delete all the team user use-cases related interfaces and follow this approach by using a generic interface. We don't need a specific interface because all the use-cases provide the same public method (execute)